File: backend/app/schemas/news.py
from datetime import datetime

from pydantic import BaseModel


class NewsImageRead(BaseModel):
    id: int
    image_url: str
    is_primary: bool
    sort_order: int


class NewsListItem(BaseModel):
    id: int
    title: str
    slug: str
    category: str
    summary: str
    primary_image_url: str | None
    published_at: datetime


class NewsListResponse(BaseModel):
    items: list[NewsListItem]
    page: int
    page_size: int
    total: int


class NewsDetail(NewsListItem):
    content: str
    views: int
    images: list[NewsImageRead]
    updated_at: datetime

File: frontend/src/types/news.ts
export interface NewsListItem {
  id: number
  title: string
  slug: string
  category: string
  summary: string
  primary_image_url: string | null
  published_at: string
}
export interface NewsImage {
  id: number
  image_url: string
  is_primary: boolean
  sort_order: number
}
export interface NewsDetail extends NewsListItem {
  content: string
  views: number
  images: NewsImage[]
  updated_at: string
}
export interface NewsListResponse {
  items: NewsListItem[]
  page: number
  page_size: number
  total: number
}
export interface NewsFilters {
  page?: number
  page_size?: number
  search?: string
  category?: string
}